scaladoc/src/dotty/tools/scaladoc/snippets/SnippetCompilerArgs.scala

```diff
@@ -1,80 +1,72 @@
 package dotty.tools.scaladoc
 package snippets
 
-case class SnippetCompilerArg(flags: Set[SCFlags]):
-  def is(flag: SCFlags): Boolean = flags.contains(flag)
+import java.nio.file.Path
 
-object SnippetCompilerArg:
-  def default: SnippetCompilerArg = SnippetCompilerArg(
-    Set(SCFlags.NoCompile)
-  )
+case class SnippetCompilerArg(flag: SCFlags, debug: Boolean):
+  def overrideFlag(f: SCFlags): SnippetCompilerArg = copy(flag = f)
 
-enum SCFlags(val flagName: String, val forbiddenFlags: Set[SCFlags]):
-  case Compile extends SCFlags("compile", Set())
-  case NoCompile extends SCFlags("nocompile", Set())
+sealed trait SCFlags(val flagName: String)
+
+object SCFlags:
```

Comment:
Now it can be enum, but I've got problems with enum before, when I wanted to keep noncompatible flags in Set.
This code doesn't compile:
```Scala
enum Foo(set: Set[Foo]):
    case Bar extends Foo(Set(Foo.Baz))
    case Baz extends Foo(Set(Foo.Bar))
```

Reply:
Hmmm, it may be by design. Value cases are translated into `val`s in companion object, so it can throw something about recursive declarations, i guess.